harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -29,14 +29,70 @@
     pass
 
 
+class TrainUnit:
+    def __init__(self, value):
+        self.value = value
+
+
+class Epoch(TrainUnit):
+    pass
+
+
+class Batch(TrainUnit):
+    pass
+
+class ShouldExit(Exception):
+    """
+    ShouldExit breaks out of the top-level workload sequencer loop from inside function calls.
+    """
+
+    def __init__(self, skip_exit_checkpoint: bool = False):
+        self.skip_exit_checkpoint = skip_exit_checkpoint
+
+    pass
+
+
+class SavableState:
+    def __init__(
+        self,
+        trial_id: int,
+        last_ckpt: int = 0,
+        steps_completed: int = 0,
+        step_id: int = 0,
+        last_val: int = 0,
+    ) -> None:
+        # Store TrialID to distinguish between e.g. pause/restart and continue training.
+        self.trial_id = trial_id
+        self.last_ckpt = last_ckpt
+        self.steps_completed = steps_completed
+        self.step_id = step_id
+        self.last_val = last_val
+
+
 class PyTorchTrialController(det.TrialController):
     def __init__(self, trial_inst: det.Trial, *args: Any, **kwargs: Any) -> None:
         super().__init__(*args, **kwargs)
 
         check.is_instance(trial_inst, PyTorchTrial, "PyTorchTrialController needs an PyTorchTrial")
         self.trial = cast(PyTorchTrial, trial_inst)
         self.context = cast(pytorch.PyTorchTrialContext, self.context)
+        self.core_context = self.context._core
```

Comment:
This is like, one character of savings.  I don't think that's worth introducing an alias.

Reply:
eh, ok.